src/server/req_runjob.c

```diff
@@ -229,29 +228,43 @@ check_and_provision_job(struct batch_request *preq, job *pjob, int *need_prov)
 static void
 clear_from_defr(int sd)
 {
+	pbs_sched *psched;
+	pbs_list_head *deferred_req;
 	struct deferred_request *pdefr;
 
-	for (pdefr = (struct deferred_request *) GET_NEXT(svr_deferred_req);
-	     pdefr;
-	     pdefr = (struct deferred_request *) GET_NEXT(pdefr->dr_link)) {
-		if (pdefr->dr_preq != NULL) {
-			if (pdefr->dr_preq->rq_conn == sd) {
-				/* found deferred run job request whose */
-				/* connection to the client has closed  */
-				if (pdefr->dr_sent != 0) {
-					/* request sent to scheduler, wait   */
-					/* for it to respond before removing */
-					/* this request, just null the qrun  */
-					/* request pointer                   */
-					pdefr->dr_preq = NULL;
-				} else {
-					/* unlink & free the deferred request */
-					delete_link(&pdefr->dr_link);
-					free(pdefr);
+	for (psched = (pbs_sched *) GET_NEXT(svr_allscheds);
+	     psched;
+	     psched = (pbs_sched *) GET_NEXT(psched->sc_link)) {
+
+		deferred_req = get_sched_deferred_request(psched, FALSE);
```

Comment:
If the second argument only passess TRUE and FALSE, consider changing it to a bool.

Reply:
Thank you. I forgot we included the `stdbool.h`. Done.